### PR TITLE
[docs]: Update README with contribution quick reference

### LINKED ISSUE #118: [FEATURE REQUEST] - Update README with a quick reference section for contributors
### Is your feature request related to a problem? Please describe.
Onboarding new contributors to this project very quickly requires a quick reference section. 

### Describe the solution you'd like
A section in the README summarizing how to contribute to the project

### PR DESCRIPTION
**Pull request by dave-ok**
**Before submitting your PR for review**

- Run `npm run lint` to find errors in code syntax/format
- Run `npm run lint:fix` to fix all auto-fixable errors in source code and auto-format with prettier
- Ensure you fix any linting errors displayed after running any of the above commands

**What does this PR do?**

Fixes #118 

Add contributions quick reference to README

**What is the link to the issue on Github?**

[Issue #118](https://github.com/microapidev/comment-microapi/issues/118)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,13 +6,19 @@ This is a microservice that allows user to create comments, edit comments and al
 
 For quick navigation:
 
+- [How it Works](#how-it-works)
```
Comment (simply-alliv) [inline README.md:9]: What is the reason for this?

File: README.md [hunk 2]
```diff
@@ -6,13 +6,19 @@ This is a microservice that allows user to create comments, edit comments and al
...
 - [Getting Started](#getting-started)
 - [Linting Your Code](#linting-your-code)
 - [Testing Your Code](#testing-your-code)
 - [Contributing Your Code](#contributing-your-code)
 - [Endpoints Documentation](#endpoints-documentation)
 - [Schema Design Explanation](#schema-design-explanation)
 
+
+## <a name="how-it-works"></a> How it works
```
Comment (simply-alliv) [inline README.md:18]: This whole section is empty so I don't see why it was added.

File: README.md [hunk 3]
```diff
@@ -44,6 +50,37 @@ You will first need to setup your local environment and ensure that all configur
...
+
+### Quick Reference
+
+- All responses must follow the format specified in the online [swagger documentation](https://comments-microservice.herokuapp.com/). To facilitate this a few utility functions have been provided and you should use them to send your responses to stay consistent. 
+   + **responseHandler** for sending responses with custom response code, optional data and a message 
+   + **customError** for error responses with custom error response code, message and optional data. To use this in a try..catch block, create a new instance of the customError class with the above properties. Pass this new error object to the `next` callback in your controller znd that is it. An **errorHandler** utility will send the necessary response
+
+- A validation middleware is created to validate all endpoints using the `Joi` validation package. Validation rules are available for every endpoint and you must use a validation middleware when implementing an endpoint. If the validation rule/middleware for the endpoint you are not working on has not been implemented then contact the maintainers.
```
Comment (simply-alliv) [inline README.md:68]: I don't think that they should contact us as maintainers. Rather encourage them to request assignment to a validation rule task that adds validation rules for the endpoint.

File: README.md [hunk 4]
```diff
@@ -44,6 +50,37 @@ You will first need to setup your local environment and ensure that all configur
...
+   + **responseHandler** for sending responses with custom response code, optional data and a message 
+   + **customError** for error responses with custom error response code, message and optional data. To use this in a try..catch block, create a new instance of the customError class with the above properties. Pass this new error object to the `next` callback in your controller znd that is it. An **errorHandler** utility will send the necessary response
+
+- A validation middleware is created to validate all endpoints using the `Joi` validation package. Validation rules are available for every endpoint and you must use a validation middleware when implementing an endpoint. If the validation rule/middleware for the endpoint you are not working on has not been implemented then contact the maintainers. 
+
+- Some endpoints are guarded and require authenication via JWT tokens. These endpoints are indicated in the swagger documentation. Authentication middleware have been provided for these endpoints. Make use of appropriate auth middleware, where required, when implementing these endpoints.
```
Comment (simply-alliv) [inline README.md:70]: There's a typo. The third sentence should use "has" instead of "have".

File: README.md [hunk 5]
```diff
@@ -44,6 +50,37 @@ You will first need to setup your local environment and ensure that all configur
...
+
+- To enable this project move swiftly. In the initial stages, tests were not written alongside implementation. For this reason, tests and implementation are separate. To enable tests not fail for non-exiatent endpoints, a helper method, **describeIfEndpoint**, has been created to defer running of tests until the endpoint itself has been created. Please use when necessary. When writing tests, for a minimum, test for a valid request's response, auth error response (if auth required), and validation error response.
```
Comment (simply-alliv) [inline README.md:72]: There's a typo in the fourth sentence, "non-exiatent" should be "non-existent".

File: README.md [hunk 6]
```diff
@@ -44,6 +50,37 @@ You will first need to setup your local environment and ensure that all configur
...
+
+- If for any reason, your contribution is impeded because some component mentioned above is not available, kindly reach out to the maintainers so that you will be advised on how to proceed.
+
+- Always lint your code using `npm run lint:fix` to catch and fix any sysntax errors, format your code with our code style choice, `prettier`. Do this also before pushing your commits to a new or an existing PR. If your code fails linting, it will not be accepted.
```
Comment (simply-alliv) [inline README.md:76]: This paragraph is redundant as there is already a section called "Linting Your Code"

File: README.md [hunk 7]
```diff
@@ -44,6 +50,37 @@ You will first need to setup your local environment and ensure that all configur
...
+
+- Ensure you rebase/sync your local repo with the latest updates from the original repo to prevent merge conflicts. Your PR will not be attended to until you have resolved all merge conflicts
+
+- When making a pull request, make use of the PR template and edit the placeholders with relevant information. PR descriptions must include reference to the issue being fixed using: fixes/resolves #issue_number e.g `fixes #12` or `resolves #25` linking to the original issue number.
+
+### Futher details
```
Comment (simply-alliv) [inline README.md:82]: Typo

File: README.md (hunk 1 above)
Comment (dave-ok) [reply README.md:9]: Will be added in a separate PR to give a big picture view of the entire micro-service. Give users/contributors an idea what the application does. I would have done that in this PR but I need to address the CommentModel first

File: README.md (hunk 2 above)
Comment (dave-ok) [reply README.md:18]: Its a placeholder

File: README.md (hunk 3 above)
Comment (dave-ok) [reply README.md:68]: I left this open to allow persons actually request the task but I'll update it to encourage them to open an issue for it and reach out if they do not understand

File: README.md (hunk 4 above)
Comment (dave-ok) [reply README.md:70]: It is "has" and not "have"

**Pull request merged**